lib/tooling/base-tool.ts

```diff
@@ -38,19 +44,21 @@ const toolCounter = new PromClient.Counter({
 });
 
 export class BaseTool {
-    constructor(toolInfo, env) {
+    protected tool: ToolInfo;
+    private env: ToolEnv;
+    private addOptionsToToolArgs = true;
+
+    constructor(toolInfo: ToolInfo, env: ToolEnv) {
         this.tool = toolInfo;
         this.env = env;
-        this.tool.exclude = this.tool.exclude ? this.tool.exclude.split(':') : [];
         this.addOptionsToToolArgs = true;
-        this.parseOutput = utils.parseOutput;
```

Comment:
parseOutput is now a protected member function, which can be overriden by subclasses if needed